users.js

```diff
@@ -0,0 +1,208 @@
+            "name": "svelikov",
+            "email": "svelikov@gmail.com",
+            "age": 20,
+            "address": "u tqh"
+        }
+    ]
+}`;
+
+//Using this object as a reference to the table rows.
+let trMap = {};
+
+const nameRegex = /[A-za-z]{1,30}/;
+const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+const addressRegex = /\w{1,300}/;
+
+if (!localStorage.getItem("database")) {
+  localStorage.setItem("database", database);
+}
+
+let tableBody = document
+  .getElementById("usersTable")
+  .getElementsByTagName("tbody")[0];
+
+const getUsers = async () => {
+  let usersJSON = localStorage.getItem("database") || database;
+  return JSON.parse(usersJSON).users;
+};
+
+const addUserToTable = (user) => {
+  let tableRow = document.createElement("tr");
+
+  let idTd = document.createElement("td");
+  let nameTd = document.createElement("td");
+  let emailTd = document.createElement("td");
+  let ageTd = document.createElement("td");
+  let addressTd = document.createElement("td");
+  let editTd = document.createElement("td");
+  let deleteTd = document.createElement("td");
+
+  let editButton = document.createElement("button");
+  let deleteButton = document.createElement("button");
+
+  editButton.innerHTML = "Edit";
+  editButton.className = "button";
+  editButton.addEventListener("click", () => loadUserToForm(user));
+
+  deleteButton.innerHTML = "Delete";
+  deleteButton.className = "button";
+  deleteButton.addEventListener("click", () => deleteUser(user));
+
+  editTd.appendChild(editButton);
+  deleteTd.appendChild(deleteButton);
+
+  idTd.innerHTML = user.id;
+  nameTd.innerHTML = user.name;
+  emailTd.innerHTML = user.email;
+  ageTd.innerHTML = user.age;
+  addressTd.innerHTML = user.address;
+
+  tableRow.appendChild(idTd);
+  tableRow.appendChild(nameTd);
+  tableRow.appendChild(emailTd);
+  tableRow.appendChild(ageTd);
+  tableRow.appendChild(addressTd);
+  tableRow.appendChild(editTd);
+  tableRow.appendChild(deleteTd);
+
+  tableBody.appendChild(tableRow);
+
+  trMap[user.id] = {
+    name: nameTd,
+    email: emailTd,
+    age: ageTd,
+    address: addressTd,
+    row: tableRow,
+  };
+};
+
+const addUserToJSON = (user) => {
+  let database = getDb();
+  let users = database.users;
+  users.push(user);
+  localStorage.setItem("database", JSON.stringify({ ...database, users }));
+};
+
+getUsers().then((usersObj) => {
+  usersObj.forEach(addUserToTable);
+});
+
+let form = document.getElementById("form");
+
+form.addEventListener("submit", (ev) => {
+  ev.preventDefault();
+
+  let id = ev.target.id.value;
+
+  let newId = (Math.random() * 10).toString().replace(".", "").substr(0, 10);
+  let name = ev.target.name.value;
+  let email = ev.target.email.value;
+  let age = ev.target.age.value;
+  let address = ev.target.address.value;
+
+  if (!nameRegex.test(name)) {
+    alert(
```

Comment:
Showing validation messages in alert is not very user friendly. Also your approach currently shows one error at a time even though I may have errors in multiple fields. I should fix current error, press submit, fix next one, etc. which is inconvenient.